pkg/virt-controller/watch/pool/pool.go

```diff
@@ -1506,3 +1549,110 @@ func (c *Controller) handleResourceUpdate(pool *poolv1.VirtualMachinePool, vm *v
 	c.recorder.Eventf(pool, k8score.EventTypeNormal, common.SuccessfulDeleteVirtualMachineReason, "Successfully updated resource %s/%s", vm.Namespace, vm.Name)
 	return nil
 }
+
+func patchFinalizer(oldFinalizers, newFinalizers []string) ([]byte, error) {
+	return patch.New(
+		patch.WithTest("/metadata/finalizers", oldFinalizers),
+		patch.WithReplace("/metadata/finalizers", newFinalizers)).
```

Comment:
Why should we patch the entire finalizers list? can't we play with the `VirtualMachinePoolControllerFinalizer` only?
This will reduce the conflicts and we can drop several code like `removePoolControllerFinalizer` function.

Reply:
Right! I got confused with the labels...